### DISCUSSION TITLE
New RP2040/Raspberry Pi Pico USB Host MSC Demo

### DISCUSSION THREAD

**Discussion by rppicomidi** [hathach/tinyusb#1726] [Show and tell]
Thank you @hathach for merging in the new host bulk endpoint support for RP2040. To try it out, I created a [demo](https://github.com/rppicomidi/pico-usb-host-msc-demo) of Mass Storage Class integrated with FatFs and a Command Line Interpreter. It does not support CDC or HID like the demo code in examples directory. It supports up to 4 USB flash drives on a USB hub. The command line interpreter implements common commands like `ls`, `rm`, `mkdir`, `cp`, etc. If you want me to make it more generic to include as an example for tinyusb, please let me know and I will have a go at it.

**Comment by rppicomidi** [comment]
I have added new code [here](https://github.com/rppicomidi/pico-usb-host-msc-demo/tree/main/PIO-USB-Demo) to allow you to run the
MSC demo using PIO USB.

**Comment by hathach** [comment]
Thank you, I am currently out of town til next week. Will check this out when backing at my desk. I have some local file system navigation written a while ago as well. We can combine them together.
